internals/secrethub/app.go

```diff
@@ -101,66 +100,48 @@ func NewApp() *App {
 		logger:          cli.NewLogger(),
 	}
 
+	commandClause := &cli.CommandClause{
+		Cmd: app.cli.Cmd,
+		App: app.cli,
+	}
+
+	app.cli.Cmd.SetUsageFunc(func(command *cobra.Command) error {
+		err := cli.Tmpl(os.Stdout, cli.UsageTemplate, commandClause)
+		if err != nil {
+			fmt.Println(err)
+		}
+		return err
+	})
+	app.cli.Cmd.SetHelpFunc(func(command *cobra.Command, i []string) {
+		err := cli.Tmpl(os.Stdout, cli.HelpTemplate, commandClause)
+		if err != nil {
+			fmt.Println(err)
+		}
+	})
 	RegisterDebugFlag(app.cli, app.logger)
 	RegisterMlockFlag(app.cli)
 	RegisterColorFlag(app.cli)
 	app.credentialStore.Register(app.cli)
```

Comment:
None of these flags seem to be registered.
You can verify this by running e.g. `secrethub tree something/something --no-color` and seeing whether it outputs the tree without colors or errors saying `unknown flag: --no-color`.

Reply:
Once this is fixed we can also test whether the credential store is working properly.

Reply:
It also seems that these "configuration" issues can be pretty hard to spot, I only stumbled upon this because you asked me to test the credential store. We should test the cli more thoroughly, any ideas on how to achieve that?

Reply:
I usually tested local flags, but never the global ones (or persistent, as cobra names them). I will think if we need to make tests that check if the CLI works as expected (as in the command is created, the flags work properly).

Reply:
Solved. Now all global flags should work as expected.

Reply:
The `--credential` flag seems to work, but the `SECRETHUB_CREDENTIAL` env var does not.

Reply:
Solved

Reply:
Fixed by #362